kosmorrolib/events.py

```diff
@@ -245,6 +247,34 @@ def _search_moon_perigee(start_time: Time, end_time: Time, timezone: int) -> [Ev
     return events
 
 
+def _search_earth_season_change(start_time: Time, end_time: Time, timezone: int) -> [Event]:
+    """Function to find earth season change event.
+
+    **Warning:** this is an internal function, not intended for use by end-developers.
+
+    Will return Summer Solstice and Autumnal Equinox on respectively 2020-06-20 and 2020-09-21:
+
+    >>> season_change = _search_earth_season_change(get_timescale().utc(2020, 5, 13), get_timescale().utc(2020, 10, 14), 0)
+    >>> len(season_change)
+    2
+    >>> oppositions[0].objects[0]
+    'Summer Solstice'
+
+    Will return nothing if there is no season change event in the period of time being calculated:
+
+    >>> _search_oppositions(get_timescale().utc(2021, 3, 20), get_timescale().utc(2021, 3, 21), 0)
+    []
+    """
+    eph = api.load('de421.bsp')
+    event = []
+    t, y = almanac.find_discrete(start_time,end_time,almanac.seasons(eph))
+    for yi, ti in zip(y,t):
+        event.append(Event(
+            EventType[f'{almanac.SEASON_EVENTS[yi].upper().replace(" ","_")}'],
+            [almanac.SEASON_EVENTS[yi]],
+            ti.utc_iso(' '))) 
```

Comment:
The `translate_to_timezone()` function should be used instead to ensure the time we get here is correctly localized to the right timezone.

```suggestion
            translate_to_timezone(ti.utc_datetime(), timezone),
```